src/crimson/os/seastore/journal/circular_journal_space.cc

```diff
@@ -229,4 +235,35 @@ CircularJournalSpace::write_header()
   );
 }
 
+CircularJournalSpace::write_ertr::future<>
+CircularJournalSpace::initialize_journal_area()
```

Comment:
It looks unnecessary to write the whole journal area at mkfs(), which can be large and wasteful.

In segment manager, when call `JournalAllocator::open()`, a header will be written with a new random nonce as `segment_header_t::segment_nonce`, and it will also be written to each of the records as `record_group_header_t::segment_nonce`.

This is why segment manager won't have this problem by checking this nonce during replay, see the related implementation in `try_decode_records_header()` indirectly called by `SegmentManagerGroup::scan_valid_records()`.

IIUC reusing most of `SegmentManagerGroup::scan_valid_records()` is part of the plan to generalizing/reusing the existing replay logic for RBM, a lot of features are already implemented there.

Reply:
Yeah, it should be a more clever way than this. I'll rewrite this PR.